### ISSUE TITLE
Can't mount mapped drive on Windows 10

### ISSUE THREAD

**Issue by m-emelchenkov** [canonical/multipass#2061]
**Describe the bug**
Multipass cannot mount SMB shared folder mapped as network drive on Windows 10.

**To Reproduce**
I have some shared folder from NAS mapped as "M:" drive in Windows 10.
`multipass mount M: testmachine` gives error `mount failed: source "M:/" does not exist`.

**Expected behavior**
Mount it as regular folder (drive), because it's transparent to Windows (it's also includes offline files). I.e. you can mount it in WSL 2, use it in any app, etc.

**Additional info**
OS: Windows 10 20H2
multipass  1.6.2+win
multipassd 1.6.2+win
Test VM is Ubuntu 18.04 LTS.

**Comment by felixkasza** [comment]
WSL2 makes SMB redirector requests as the logged-on Windows user by brokering such requests. It uses the VM machinery only to virtualise as little as possible, which allows for a very tight integration.

Multipass itself runs as SYSTEM, i. e., without an identity that works off-box. It also cannot impersonate whatever user is logged on to your Linux VM, since those users are local to the Linux installation and have no mapping to users on the host Windows OS. 

In short, I am not sure that this can be changed without a major effort. One possibility might be a private channel from the Linux VM to multipassd, with a file system driver on the Linux guest passing the request, with authentication credentials, to multipassd for execution – which sounds (a) a lot like WSL2's brokering of Windows file system access and (b) like a lot of work. :(

I'd love to see this feature, too, but …
